### PR TITLE
Php4ConstructorFixer - fix causing infinite recursion

### LINKED ISSUE #1051: PHP4 Constructor fixer can cause infinite recursion
After some more testing, I came across a new unexpected variant, that can be seen here:

https://github.com/revive-adserver/revive-adserver/blob/79e7020df28d81e77190fb6355141cb6fdd1a20b/lib/pear/MDB2/Schema/WriterChangeset.php#L32

Basically a class that extends another one, with a PHP4 constructor invoking the parent constructor as ```$this->__construct()```.

The fixer currently converts it to:

https://github.com/revive-adserver/revive-adserver/blob/30caba8170c722fa82c6cc9c3f13d8152c3220ee/lib/pear/MDB2/Schema/WriterChangeset.php#L32

I'll see what I can do in the next few days / weeks ;)

### PR DESCRIPTION
**Pull request by mbeccati**
Fix #1051

### CODE REVIEW

File: Symfony/CS/Tests/Fixer/Contrib/Php4ConstructorFixerTest.php [hunk 1]
```diff
@@ -981,4 +981,51 @@ function __construct($foo)
...
+        }
+    }
+?>
+EOF;
+
+        $input = <<<'EOF'
+<?php
+    class Parent1
+    {
+        function _construct()
```
Comment (GrahamCampbell) [inline Symfony/CS/Tests/Fixer/Contrib/Php4ConstructorFixerTest.php]: Is this typo intentional?

Comment (mbeccati) [inline Symfony/CS/Tests/Fixer/Contrib/Php4ConstructorFixerTest.php]: Definitely not. The definition of the parent class is just decorative here, but still worth fixing.

Comment (GrahamCampbell) [inline Symfony/CS/Tests/Fixer/Contrib/Php4ConstructorFixerTest.php]: It occurs in two places btw. ;)

Comment (keradus) [inline Symfony/CS/Tests/Fixer/Contrib/Php4ConstructorFixerTest.php]: ping @mbeccati

**Comment by mbeccati** [comment]
pong @keradus ;)

Sorry too much on my plate and I couldn't take care of it until now.

**Comment by keradus** [comment]
:+1:

**Comment by GrahamCampbell** [comment]
:+1:

**Comment by keradus** [comment]
Thank you @mbeccati.

**Pull request merged**